scalameta/quasiquotes/shared/src/main/scala/scala/meta/internal/quasiquotes/ReificationMacros.scala

```diff
@@ -122,10 +126,7 @@ class ReificationMacros(val c: Context) extends AstReflection with AdtLiftables
         }
         curr + 1
       }
-      val result = {
-        if (reflectInput.file.file != null) Input.File(reflectInput.file.file)
-        else Input.String(new String(reflectInput.content)) // NOTE: can happen in REPL or in custom Global
-      }
+      val result = Input.String(new String(reflectInput.content))
```

Comment:
I added a new `Input.VirtualFile` which is an `AbsolutePath` along with its file contents. That solves the linking problem in Scala.js. I imagine this will be useful other contexts such as for example browser extensions to parse files.